core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSAction.java

```diff
@@ -0,0 +1,126 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Move file(s) within HDFS.
+ * A user must specify file/directory path and destination file/directory path
+ * Optionals include fileRegex
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("HDFSAction")
+@Description("Action to move files within HDFS")
+public class HDFSAction extends Action {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSAction.class);
+
+  private HDFSActionConfig config;
+
+  public HDFSAction(HDFSActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    Path source = new Path(config.sourcePath);
+    Path dest = new Path(config.destPath);
+    FileSystem fileSystem = source.getFileSystem(new Configuration());
+
+    if (fileSystem.getFileStatus(source).isFile()) { //moving single file
+
+      try {
+        fileSystem.rename(source, dest);
+      } catch (IOException e) {
+        LOG.error("Failed to move file {} to {} for reason: {}", source.toString(), dest.toString(), e);
+      }
+      return;
+    }
+
+    //moving contents of directory
+
+    FileStatus[] listFiles;
+    if (config.fileRegex != null) {
+      PathFilter filter = new PathFilter() {
+        private final Pattern pattern = Pattern.compile(config.fileRegex);
+
+        @Override
+        public boolean accept(Path path) {
+          return pattern.matcher(path.getName()).matches();
+        }
+      };
+
+      listFiles = fileSystem.listStatus(source, filter);
+    } else {
+      listFiles = fileSystem.listStatus(source);
+    }
+
+    for (FileStatus file: listFiles) {
+      source = file.getPath();
+      //Moving directory, so destination filePath doesn't have filename
+      dest = new Path(config.destPath);
```

Comment:
What about the case of moving just a single file?

On Aug 5, 2016 9:43 AM, "Terence Yim" <notifications@github.com> wrote:

> In core-plugins/src/main/java/co/cask/hydrator/plugin/batch/
> action/HDFSAction.java
> <https://github.com/caskdata/hydrator-plugins/pull/357#discussion_r73722833>
> :
>
> > +
> > +        @Override
> > +        public boolean accept(Path path) {
> > +          return pattern.matcher(path.getName()).matches();
> > +        }
> > +      };
> > +
> > +      listFiles = fileSystem.listStatus(source, filter);
> > +    } else {
> > +      listFiles = fileSystem.listStatus(source);
> > +    }
> > +
> > +    for (FileStatus file: listFiles) {
> > +      source = file.getPath();
> > +      //Moving directory, so destination filePath doesn't have filename
> > +      dest = new Path(config.destPath);
>
> I think it is better if you construct the target path using the dest as
> the parent and the source name as the child
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/caskdata/hydrator-plugins/pull/357/files/64c17e86da0e2d0f18487b23a1146c27b857e8fa#r73722833>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACU_ETWk6caJx_JxGe7T8OWfNc82AW_4ks5qc2g3gaJpZM4JcA2r>
> .
>